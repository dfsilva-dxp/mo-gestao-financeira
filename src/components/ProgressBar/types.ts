import { BalanceCardProps } from "components/BalanceCard/types";

export type ProgressPropsProps = Pick<
  BalanceCardProps,
  "progressBar" | "color"
>;
