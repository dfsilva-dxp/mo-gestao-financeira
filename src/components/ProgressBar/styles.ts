import styled, { css, DefaultTheme } from "styled-components";
import { ProgressPropsProps } from "./types";

export const Bar = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.xxsmall};
    width: 100%;
  `}
`;

const progressModifiers = {
  primary: (theme: DefaultTheme) => css`
    background: ${theme.colors.primary};
  `,
  secondary: (theme: DefaultTheme) => css`
    background: ${theme.colors.secondary};
  `
};

type ProgressProps = ProgressPropsProps;

export const Progress = styled.div<ProgressProps>`
  ${({ theme, progressBar, color }) => css`
    background: ${theme.colors.black};
    width: 100%;
    height: 1rem;
    border-radius: ${theme.border.radius.default};
    position: relative;

    &::after {
      content: "";
      background: ${theme.colors.secondary};
      width: ${`${progressBar}%`};
      height: 1rem;
      border-radius: ${theme.border.radius.default};
      position: absolute;

      ${!!color && progressModifiers[color](theme)}
    }
  `}
`;
