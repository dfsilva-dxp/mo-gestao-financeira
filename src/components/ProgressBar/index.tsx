import { Bar, Progress } from "./styles";
import { ProgressPropsProps } from "./types";

export default function ProgressBar({
  progressBar,
  color = "primary"
}: ProgressPropsProps) {
  return (
    <>
      {!!progressBar && (
        <Bar>
          <Progress
            progressBar={progressBar}
            color={color}
            role="progressbar"
            aria-label="Progress Bar"
          />
        </Bar>
      )}
    </>
  );
}
