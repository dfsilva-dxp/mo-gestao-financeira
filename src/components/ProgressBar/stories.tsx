import { Story, Meta } from "@storybook/react/types-6-0";
import { ComponentProps } from "react";

import ProgressBar from ".";

export default {
  title: "Components/ProgressBar",
  component: ProgressBar,
  argTypes: {
    color: {
      options: ["primary", "secondary"],
      control: {
        type: "inline-radio"
      }
    }
  },
  args: {
    progressBar: 80,
    color: "primary"
  },
  parameters: {
    layout: "centered"
  },
  decorators: [
    (Story) => (
      <div style={{ margin: "0 auto", width: "47.6rem" }}>
        <Story />
      </div>
    )
  ]
} as Meta;

export const Default: Story<ComponentProps<typeof ProgressBar>> = ({
  ...props
}) => <ProgressBar {...props} />;
