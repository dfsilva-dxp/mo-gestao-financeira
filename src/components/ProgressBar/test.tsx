import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import ProgressBar from ".";

const props = {
  progressBar: 80
};

describe("<ProgressBar />", () => {
  it("should render primary color by default", () => {
    const { container } = renderWithTheme(<ProgressBar {...props} />);

    expect(
      screen.getByRole("progressbar", { name: /progress bar/i })
    ).toHaveStyleRule("background", "#F9769D", {
      modifier: "::after"
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render #96A7FF when secondary color is passed", () => {
    renderWithTheme(<ProgressBar {...props} color="secondary" />);

    expect(
      screen.getByRole("progressbar", { name: /progress bar/i })
    ).toHaveStyleRule("background", "#96A7FF", {
      modifier: "::after"
    });
  });
});
