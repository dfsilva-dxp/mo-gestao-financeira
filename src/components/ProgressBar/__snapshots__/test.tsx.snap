// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> should render primary color by default 1`] = `
.c0 {
  margin-top: 0.8rem;
  width: 100%;
}

.c1 {
  background: #030517;
  width: 100%;
  height: 1rem;
  border-radius: 1.6rem;
  position: relative;
}

.c1::after {
  content: "";
  background: #96A7FF;
  width: 80%;
  height: 1rem;
  border-radius: 1.6rem;
  position: absolute;
  background: #F9769D;
}

<div
  class="c0"
>
  <div
    aria-label="Progress Bar"
    class="c1"
    color="primary"
    role="progressbar"
  />
</div>
`;
